packages/components/src/UISwitch.test.tsx

```diff
@@ -1,34 +1,34 @@
 import React from 'react';
 import { render } from '@testing-library/react';
 import userEvent from '@testing-library/user-event';
-import UISwitch from './UISwitch';
+import UISwitch, { UISwitchProps } from './UISwitch';
 
-function makeUISwitch(testId: string, onClick) {
-  return render(<UISwitch on onClick={onClick} data-testid={testId} />);
+function makeUISwitch({
+  'data-testid': dataTestId = 'TestUISwitch',
+  onClick = jest.fn(() => true),
+  on = true,
+}: Partial<UISwitchProps> = {}) {
+  return render(
+    <UISwitch on={on} onClick={onClick} data-testid={dataTestId} />
+  );
 }
 
 it('mounts and unmounts properly', () => {
-  function click() {
-    return true;
-  }
-
-  const testId = 'test id for UISwitch';
-  const testSwitch = makeUISwitch(testId, click);
-  testSwitch.unmount();
+  makeUISwitch();
 });
 
 it('get element by data-testid works', () => {
-  let result = false;
-  const changeFalseToTrue = () => {
-    result = true;
+  let clicked = false;
+  const onClick = () => {
+    clicked = true;
   };
   const testId = 'test id for UISwitch';
-  const testSwitch = makeUISwitch(testId, changeFalseToTrue);
+  const testSwitch = makeUISwitch({ 'data-testid': testId, onClick });
   const elements = testSwitch.getAllByTestId(testId);
   expect(elements.length).toBe(1);
   const button = elements[0];
   expect(button instanceof HTMLButtonElement).toBe(true);
-  expect(result).toBe(false);
+  expect(clicked).toBe(false);
   userEvent.click(button);
-  expect(result).toBe(true);
+  expect(clicked).toBe(true);
```

Comment:
Reading this again - all you're really doing here is checking that the method is called when the button is clicked. You could simply check that the jest fn was called instead of having your own intermediate variable, for example:
```
it('get element by data-testid works', () => {
  const onClick = jest.fn();
  const testId = 'test id for UISwitch';
  const testSwitch = makeUISwitch({ 'data-testid': testId, onClick });
  const elements = testSwitch.getAllByTestId(testId);
  expect(elements.length).toBe(1);
  const button = elements[0];
  expect(button instanceof HTMLButtonElement).toBe(true);
  expect(onClick).not.toHaveBeenCalled();
  userEvent.click(button);
  expect(onClick).toHaveBeenCalledTimes(1);
});
```